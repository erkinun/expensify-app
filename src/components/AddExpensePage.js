import React from 'react';
import ExpenseForm from './ExpenseForm'

const AddExpensePage = () => (
    <div>
        <h1>Expense Add</h1>
        <ExpenseForm />
    </div>
);

export default AddExpensePage