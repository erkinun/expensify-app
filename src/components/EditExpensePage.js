import React from 'react';
import { connect } from 'react-redux'

// edit page
const EditExpensePage = (props) => {
    console.log(props)
    return (
        <div>
            This is from my edit expense component for {props.match.params.id}
        </div>
    )
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find(e => e.id === props.match.params.id)
    }
}

export default connect(mapStateToProps)(EditExpensePage)