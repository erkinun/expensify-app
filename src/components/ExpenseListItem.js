import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

export const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div>
            <h3 className="title">{description}</h3>
            <span className="subtitle">{moment(createdAt).format('Do MMMM YYYY')}</span>
        </div>
        <h3 className="data">{numeral(amount / 100).format('$0,0.00')} </h3>
    </Link>
)
