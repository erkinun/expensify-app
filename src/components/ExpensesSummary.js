import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import expenseTotals from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = (props) => {
    const expenseCount = props.expenses.length
    const expenseWord = expenseCount > 1 ? 'expenses' : 'expense'
    const total = numeral(expenseTotals(props.expenses) / 100).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="title">Viewing <span>{expenseCount} {expenseWord}</span> totalling <span>{total}</span> </h1>
                <div className="actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)