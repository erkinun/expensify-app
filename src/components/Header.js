import React from 'react';
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth'

export const Header = ({logout}) => (
    <header>
        <h1>Expensify</h1>
        <p><NavLink to="/dashboard" exact={true} activeClassName="is-active">Dashboard</NavLink></p>
        <p><NavLink to="/create" activeClassName="is-active">Create Expense</NavLink></p>
        <p><NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink></p>
        <button onClick={logout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
  logout: () => {
      dispatch(startLogout())
  }
})

export default connect(undefined, mapDispatchToProps)(Header)