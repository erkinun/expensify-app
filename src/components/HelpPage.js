import React from 'react'

// help page
const HelpPage = () => (
    <div>
        This is from my help component
    </div>
);

export default HelpPage
