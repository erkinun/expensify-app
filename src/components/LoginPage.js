import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export class LoginPage extends React.Component {
  render() {
    return (
      <div className="box_layout">
        <div className="box">
          <h1 className="title">Expensify</h1>
          <p>It's time to track your expenses from web</p>
          <button className="button" onClick={this.props.startLogin}>Login with Google</button>
        </div>
      </div> 
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)