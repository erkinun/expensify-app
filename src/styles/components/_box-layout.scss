.box_layout {
  background: url('/images/bg.jpg');
  background-size: cover;
  height: 100vh;
  width: 100vw;
  display: flex;
  align-items: center;
  justify-content: center;

  .box {
    background: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
  }

  .title {
    margin: 0 0 $m-size 0;
    line-height: 1;
  }
}