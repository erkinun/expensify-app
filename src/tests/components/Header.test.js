import { shallow } from 'enzyme'
import React from 'react'
import { Header } from '../../components/Header'

let startLogout, wrapper

beforeEach(() => {
    startLogout = jest.fn()
    wrapper = shallow(<Header logout={startLogout}/>)
})

test('should render Header correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('should call startLogout on button click', () => {
    wrapper.find('button').prop('onClick')()
    expect(startLogout).toBeCalled()
})



