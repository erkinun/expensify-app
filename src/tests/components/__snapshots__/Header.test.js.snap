// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="content"
    >
      <Link
        className="title"
        exact={true}
        replace={false}
        to="/dashboard"
      >
        <h1>
          Expensify
        </h1>
      </Link>
      <button
        className="button link"
        onClick={[MockFunction]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
